include/mata/nfa/strings.hh

```diff
@@ -354,8 +354,11 @@ NoodleSubstSequence noodlify_mult_eps(const SegNfa& aut, const std::set<Symbol>&
  *                 minimization before noodlification.
  * @return A list of all (non-empty) noodles.
  */
-NoodleSequence noodlify_for_equation(const AutRefSequence& left_automata, const Nfa::Nfa& right_automaton,
-                                     bool include_empty = false, const StringMap& params = {{"reduce", "false"}});
+NoodleSequence noodlify_for_equation(
+        const std::vector<std::reference_wrapper<Nfa::Nfa>>& left_automata,
+        const Nfa::Nfa& right_automaton,
```

Comment:
Suggestion: maybe add `using Mata::Nfa` or something similar? The `Nfa::Nfa` looks weird.

Reply:
There is nothing we can do about it. That is the problem of #204. We could only use `Mata::Nfa::Nfa` or some strange alias such as `using NfaStruct = Mata::Nfa::Nfa`.

Reply:
@kilohsakul @jurajsic Should we think about a better name for the `Mata::Nfa` (and for string solving `Mata::Nfa::SegNfa`) namespace? Such as `Mata::NFA`, or `Mata::Nfas` (that is not idiomatic, but it would work, too) or something similar? The same name for namespace and structure causes these “problems”.

Reply:
This question is related to #204.

Reply:
Yes this is annoying. What about all namespaces with all capital letters? It is a little ugly, but one gets used to it and these problems will disappear.

Reply:
Definitely not all namespaces, but the few namespaces with conflicting names which represent an abbreviation anyway could use ALLCAPS format. That is, `NFA`, `AFA`, `SegNFA`, etc. This would solve the conflicts. `NFA::Nfa` looks good to me, too.

Alternatively, something like `Mata::Nfas::Nfa` could be used, but that is an inferior option to the all capital letters version.

Reply:
We will resolve this issue in #204. For now, I used `Mata::Nfa::Nfa` to clarify what is being created.

Reply:
ok, lets continue there